podpac/core/compositor.py

```diff
@@ -81,20 +82,51 @@ class Compositor(Node):
     source = tl.Unicode().tag(attr=True)
     sources = ArrayTrait(ndim=1)
     cache_native_coordinates = tl.Bool(True)
-
     interpolation = interpolation_trait(default_value=None)
+    strict_source_outputs = tl.Bool(False)
+
+    tl.default("source")
 
-    @tl.default("source")
     def _source_default(self):
-        source = []
-        for s in self.sources[:3]:
-            source.append(str(s))
-        return "_".join(source)
+        return "_".join(str(source) for source in self.sources[:3])
 
     @tl.default("source_coordinates")
     def _source_coordinates_default(self):
         return self.get_source_coordinates()
 
+    @tl.validate("sources")
+    def _validate_sources(self, d):
+        self.outputs  # check for consistent outputs
+        return [copy.deepcopy(source) for source in d["value"]]
```

Comment:
@mpu-creare @mlshapiro I think this is the desired behavior, but let me know if I'm mistaken. I haven't had a lot to do with interpolation. See `test_compositor::TestCompositor::test_iteroutputs_interpolation` for the case this covers.

Reply:
Got it. Nice fix. Thanks!